actors/builtin/market/market_actor.go

```diff
@@ -216,7 +219,12 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 			err = msm.dealProposals.Set(id, &deal.Proposal)
 			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to set deal")
 
-			err = msm.dealsByEpoch.Put(deal.Proposal.StartEpoch, id)
+			// We should randomize the first epoch for when the deal will be processed so an attacker isn't able to
+			// schedule too many deals for the same tick.
+			processEpoch, err := genRandNextEpoch(deal.Proposal.StartEpoch, rt.CurrEpoch(), &deal.Proposal, rt.GetRandomness)
```

Comment:
Since you pass the proposal anyway, don't bother passing the start epoch separately.

Reply:
This is done.